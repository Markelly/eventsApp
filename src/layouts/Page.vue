<template>
  <div class="view-layout">
    <div class="view-layout__header">
      <h1 data-events-title>{{ pageTitle }}</h1>
    </div>
    <div class="view-layout__body">
      <b-button class="view-layout__body__button" variant="secondary" data-favorites-button @click="goToPage">
        {{ buttonLabel }}
      </b-button>
      <slot name="content" />
    </div>
  </div>
</template>

<script>
export default {
  name: 'ViewLayout',
  props: {
    pageTitle: {
      type: String,
      required: true
    },
    buttonLabel: {
      type: String,
      required: true
    }
  },
  methods: {
    goToPage() {
      this.$emit('click')
    }
  }
}
</script>

<style scoped lang="scss">
@import '@/styles/events-app';
.view-layout {
  max-width: $max-content-size;
  margin: 24px auto;
  padding: 8px;

  &__header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    top: 0;
    left: 0;
    width: 100%;
    position: fixed;
    height: 100px;
    background: $color-white;
    z-index: 2;
    padding: 0 24px;
    box-shadow: 8px 4px 4px $color-light-green;
  }

  &__body {
    margin-top: 120px;
    display: flex;
    flex-direction: column;

    &__button {
      align-self: flex-start;
    }
  }
}
</style>
