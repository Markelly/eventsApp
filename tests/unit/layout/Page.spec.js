import { createLocalVue, shallowMount } from "@vue/test-utils";
import { BootstrapVue } from "bootstrap-vue";
import Vuex from "vuex";
import Router from "vue-router";
import Page from "@/layouts/Page";

const localVue = createLocalVue();
localVue.use(BootstrapVue);
localVue.use(Vuex);
localVue.use(Router);

describe("<Page />", () => {
  let wrapper, spyDispatch, mockRoute, store;

  beforeEach(() => {
    mockRoute = new Router({});
    store = new Vuex.Store({
      modules: {
        namespaced: true,
        actions: {
          logout: jest.fn()
        }
      }
    });
    spyDispatch = jest.spyOn(store, "dispatch");
    wrapper = shallowMount(Page, {
      localVue,
      store,
      router: mockRoute,
      propsData: {
        pageTitle: "foo",
        buttonLabel: "bar"
      }
    });
  });

  test("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("shold emit an event to move to another page", () => {
    wrapper.vm.goToPage();
    expect(wrapper.emitted("click")).toBeTruthy();
  });

  test("should logout user", () => {
    wrapper.vm.logout();
    expect(spyDispatch).toHaveBeenCalledWith("logout");
  });
});
