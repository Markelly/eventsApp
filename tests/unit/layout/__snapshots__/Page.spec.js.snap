// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should render the component 1`] = `
<div class="view-layout">
  <div class="view-layout__header">
    <h1 data-events-title="">foo</h1>
    <b-button-stub variant="primary" type="button" tag="button" target="_self">Logout</b-button-stub>
  </div>
  <div class="view-layout__body">
    <b-button-stub variant="secondary" type="button" tag="button" target="_self" data-favorites-button="" class="view-layout__body__button">
      bar
    </b-button-stub>
  </div>
</div>
`;
